src/controllers/edgeconnect/deployment/deployment.go

```diff
@@ -0,0 +1,145 @@
+package deployment
+
+import (
+	edgeconnectv1alpha1 "github.com/Dynatrace/dynatrace-operator/src/api/v1alpha1/edgeconnect"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/edgeconnect/consts"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects/address"
+	"github.com/Dynatrace/dynatrace-operator/src/webhook"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func New(instance *edgeconnectv1alpha1.EdgeConnect) *appsv1.Deployment {
+	appLabels := buildAppLabels()
+	labels := kubeobjects.MergeMap(
+		instance.Labels,
+		appLabels.BuildLabels(),
+		map[string]string{kubeobjects.AppInstanceLabel: instance.ObjectMeta.Name},
+	)
```

Comment:
if I look at other usage of `kubeobjects.NewAppLabels`:
```
appLabels := kubeobjects.NewAppLabels(kubeobjects.OneAgentComponentLabel, dynakube.Name,
		dsInfo.deploymentType, versionLabelValue)
```

we do not need the `AppInstanceLabel` as that is already the `CreatedBy` label, which is set according to the 2. param in `kubeobjects.NewAppLabels`

Reply:
@chrismuellner, we would appreciate your judgment here. I know it's a new label compared to DynaKube, but I followed the requirement spec for CR:
```
labels:
    app.kubernetes.io/instance: <.metadata.name>
```